pylint/extensions/code_style.py

```diff
@@ -141,6 +153,14 @@ def _check_inplace_defined_list(
         if isinstance(node.iter, astroid.List):
             self.add_message("consider-using-tuple", node=node.iter)
 
+    def _check_in_comparison(self, comparator: NodeNG) -> None:
+        """Checks for membership comparisons with in-place container objects."""
+        if not isinstance(comparator, _BaseContainer):
```

Comment:
I guess it's a legitimate use of `_BaseContainer` which means we should make it public in astroid ?

Reply:
Yeah, that would probably be for the best.